ui/helpers/utils/util.js

```diff
@@ -63,6 +63,30 @@ export function formatDateWithYearContext(
     now.year === dateTime.year ? formatThisYear : fallback,
   );
 }
+
+export function formatDateWithSuffix(timestamp) {
```

Comment:
I don't think this will localize, since I still see english months after switching languages.  Might want to try `Intl.DateTimeFormat`.  Similar to this snippet, but with different arguments so it's just the month/day/year.

https://github.com/MetaMask/metamask-extension/blob/7b3450a294a2fe5751726a9c33d6fa0b564f03dd/ui/pages/asset/util.ts#L3-L10